presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -4907,6 +4907,9 @@ public void testScalarSubquery()
         // exposes a bug in optimize hash generation because EnforceSingleNode does not
         // support more than one column from the underlying query
         assertQuery("SELECT custkey, (SELECT DISTINCT custkey FROM orders ORDER BY custkey LIMIT 1) FROM orders");
+
+        // cast scalar sub-query
+        assertQuery("SELECT 1.0/(SELECT 1)");
```

Comment:
you can add tests for `IN` like:

```
SELECT 1.0 IN (SELECT 1), 2.0 IN (SELECT 1) WHERE 1.0 IN (SELECT 1);
```

SemiJoin (`IN` with subquery) is tested in different method.